trie_test.go

```diff
@@ -8,6 +8,59 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+func TestTrie_Scan(t *testing.T) {
+	tri := trie.New()
+	tri.Put([]string{"d", "a", "l", "i"}, 2)
```

Comment:
Test for Walk() should assert that all (key, value) pairs were returned in-order.